test/EVMHost.cpp

```diff
@@ -449,11 +447,11 @@ evmc::result EVMHost::precompileECRecover(evmc_message const& _message) noexcept
 			}
 		}
 	};
-	evmc::result result = precompileGeneric(_message, inputOutput);
+	evmc::Result result = precompileGeneric(_message, inputOutput);
 	// ECRecover will return success with empty response in case of failure
 	if (result.status_code != EVMC_SUCCESS && result.status_code != EVMC_OUT_OF_GAS)
-//		return resultWithGas(_message.gas, gas_cost, {});
 	{
+		// Could use resultWithGas(_message.gas, gas_cost, {}) here, but it expects a reference.
```

Comment:
Looking at this here now, I actually see that ``resultWithGas`` *doesn't* copy, but stores a pointer in ``result.output_data``...
Is evmc robust against having "garbage" in ``result.output_data`` as long as ``result.output_size`` is zero?
If so, this is still fine. If not, we may actually want to set ``result.output_data`` to ``nullptr`` in ``EVMHost::resultWithGas`` if ``_data`` is empty.

But yeah, if too much hassle, we can also leave it at this comment, if need be ;-).

Reply:
https://github.com/ethereum/solidity/blob/develop/test/evmc/evmc.h#L357-L375 says that `output_data` MAY be null, and if it is, `output_size` MUST be null. Doesn't state that if `output_size` is null then `output_data` MUST NOT be accessed. @chfast do you have any comments?

Reply:
Could also change `resultWithGas` such that
```diff
-       result.output_data = _data.data();
        result.output_size = _data.size();
+       if (result.output_size == 0)
+               result.output_data = nullptr;
```

Reply:
Could also have an overload of `resultWithGas` without a `_data` field 🤓

Reply:
> Could also change `resultWithGas` such that
> 
> ```diff
>         result.output_data = _data.data();
>         result.output_size = _data.size();
> +       if (result.output_size == 0)
> +               result.output_data = nullptr;
> ```

Or
```
result.output_data = _data.empty() ? _data.data() : nullptr;
result.output_size = _data.size();
```
That's what I meant above. Actually unfortunate in the C++ standard that ``.data()`` specs as "If size() is ​0​, data() may or may not return a null pointer.", but what can one do...

Reply:
EVM will handle non-null pointer without problem so `_data.data(), _data.size()` should always be fine.

Reply:
> EVM will handle non-null pointer without problem so _data.data(), _data.size() should always be fine.

Can we put the requirement in writing in `evmc.h` not to dereference `output_data` if `output_size` is 0?

Reply:
Yes, you can put it there.

Reply:
Added in https://github.com/ethereum/evmc/pull/673.

@ekpyron do you still want to add the change or can we merge?

Reply:
Given https://github.com/ethereum/evmc/pull/673, I'd say let's switch to the currently commented version here.

Reply:
Updated, @ekpyron please review.